op-e2e: Add action test for consolidation step prior to interop being active

ajsutton opened the pull request:
**Description**

Mostly checks the supervisor can return super roots prior to interop. The action test deployment is fixed to not include the interop contracts in L2 genesis unless interop is enabled from genesis.  The test sends a message to the CrossL2Inbox which reverts because the proxy is uninitialised (because we didn't deploy the contract yet). The test then confirms the transaction isn't re-orged out (with revert status) and that op-program matches the op-supervisor behaviour.

**Metadata**

part of #15625

ajsutton committed:
op-e2e: Add action test for consolidation step prior to interop being active.

Mostly checks the supervisor can return super roots prior to interop.

tynes commented:
Historically we have always been backwards and forwards compatible with L2 genesis states, if we must not have the `CrossL2Inbox` deployed if interop isn't active, then we will no longer be backwards compatible or the way to frame it is that interop messages are not validated until the interop hardfork is live

ajsutton commented:
> Historically we have always been backwards and forwards compatible with L2 genesis states, if we must not have the `CrossL2Inbox` deployed if interop isn't active, then we will no longer be backwards compatible or the way to frame it is that interop messages are not validated until the interop hardfork is live

@tynes Hmm, I've heard the assumption that executing messages won't be possible at all prior to interop a number of times and I don't believe we have any testing for how things work if they do occur.  It's reasonably easy to skip checking messages in the consolidate step in proofs (and I was going to do that for safety, though it's not quite as straight forward as I'd hoped), but we would need to ensure op-supervisor/op-node also skips them rather than just assuming they can't happen. They may work already but I don't recall seeing any tests for it.

ajsutton committed:
op-e2e: Include invalid executing message prior to interop activating.

Currently fails because both op-supervisor and op-program incorrectly re-org out the invalid message.

ajsutton commented:
Ok, I can confirm that op-supervisor incorrectly re-orgs out invalid executing messages even if the interop hard fork is not yet active.  Updated the test to include an invalid message to show this problem.

ajsutton committed:
op-e2e: Don't deploy interop contracts in L2 genesis unless interop is active.

Update pre-activation test to expect the executing message to revert.
